src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,43 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int AGE_18 = 18;
+    private static final int SIX_SYMBOLS = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RuntimeException("Invalid input user");
+        }
+        if (user.getLogin() == null) {
+            throw new RuntimeException("Invalid login");
+        }
+        String pureLogin = user.getLogin().trim();
+        user.setLogin(pureLogin);
+        if (user.getLogin().isEmpty()) {
+            throw new RuntimeException("Empty login");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("Login already exists");
```

Comment:
```suggestion
            throw new RuntimeException("Login already exists " + user.getLogin());
```

let's make exception messages more informative (fix in all places)